tf-modules/cumulus/variables.tf

```diff
@@ -100,6 +100,12 @@ variable "elasticsearch_security_group_id" {
   default     = ""
 }
 
+variable "lambda_timeouts" {
+  description = "Configurable map of timeouts for ingest task lambdas in the form <lambda_identifier>_timeout: <timeout>"
+  type = map(string)
```

Comment:
Do we want to use a type of `object` instead where we explicitly enumerate the allowed properties? The difference is that using `map(string)` means that someone can deploy by sending an object with properties that aren't supported and TF won't complain, but if they misspelled something, it would also have no effect and possibly be confusing. Whereas by using an object with explicitly enumerated properties, TF will complain if you try to send an object with unsupported properties (I believe)

Reply:
If we really are going to allow memory overrides, I would kind of prefer one mapping to manage everything, but that's probably getting into premature optimization territory

And yeah, I forgot that an object would require you to specify all the values. Maybe that's a worthy tradeoff for getting validation of your object properties? Otherwise it's also hard even for **us** as future maintainers to know all of the properties supported by this mapping without digging through the code to see how it is used.

Also, do we need something in the user facing docs for this?

Reply:
>  I would kind of prefer one mapping to manage everything, 

Agreed, but that would require an entirely separate scope of work, and more importantly be a bit fragile to adding keys as users would have to define all keys for each lambda they want to configure a particular value for.    If you get much greater than 3 config values, that's pretty mediocre as well.       I'm tempted to make timeout/memory as a pair, but I can't come up with a good justification for it other than personal preference. 

> Also, do we need something in the user facing docs for this?

A good point, however to some degree the configuration variables *are* the docs per the module README. 

We could add a section under `Configuration` for task configurations if we think we'll be doing more of this.   At a minimum the variable descriptions need updated, will sort that in a moment.

Reply:
How about: 303786860 ?  

I don't love it,but I love not having a doc even less.